src/Agent.Worker/TestResults/TestRunDataPublisherHelper.cs

```diff
@@ -0,0 +1,170 @@
+                                if (_testRunPublisher.IsTestRunFailed(testRunStats, cancellationToken))
+                                {
+                                    _executionContext.Output("Failed Results are published");
+                                    return true;
+                                }
+                            }
+                            catch (AggregateException exception)
+                            {
+                                _executionContext.Output($"TestRunDataPublisher.DoesRunsContainsFailures: Exception occured using test run summary api {exception}");
+                                //After the max retries, if API still fails to calculate the summary and return with exception
+                                //It will fallback to run stats api which may result in dirty summary i.e. old flow.
+                                if (exception.InnerException is TestObjectNotFoundException)
+                                {
+                                    _executionContext.Warning($"TestRunDataPublisher.DoesRunsContainsFailures: Validating if test run is failed or succeeded based on summary where calulation of summary is not in completed state.");
+                                    return _testRunPublisher.IsTestRunFailed(projectName, testRun.Id, cancellationToken);
+                                }
+                                else
+                                {
+                                    _executionContext.Error($"TestRunDataPublisher.DoesRunsContainsFailures: Not retriable exception occured, hence throwing the exception: {exception}");
+                                    throw;
+                                }
+                            }
+                        }
+                        else
+                        {
+                            if (_testRunPublisher.IsTestRunFailed(projectName, testRun.Id, cancellationToken))
+                            {
+                                _executionContext.Output("Failed Results are published");
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                return false;
+            }
+            catch (Exception ex)
+            {
+                _executionContext.Output($"Failed to get test run outcome from publisher: {ex}");
+                return null;
+            }
+        }
+
+        internal protected virtual string GetEnvironmentVar(string envVar)
+        {
+            return Environment.GetEnvironmentVariable(envVar);
+        }
+
+        internal protected virtual CancellationToken GetCancellationToken()
+        {
+            return new CancellationToken();
+        }
+
+        protected internal virtual bool GetFeatureFlagState(string featureFlagName, Guid serviceInstaceGuid)
```

Comment:
There's already a FF check in Utils in here.

Reply:
@navin22 It is using the one from util's FeatureFlagService. A wrapper over util's GetFeatureFlagState method for this class.